src/NuGet.Common/ConcurrencyUtilities.cs

```diff
@@ -10,41 +10,35 @@ namespace NuGet.Common
 {
     internal static class ConcurrencyUtilities
     {
-        internal static async Task<T> ExecuteWithFileLocked<T>(string filePath, Func<Task<T>> action)
+        public static Task<TVal> ExecuteWithFileLocked<TVal>(string filePath, TimeSpan timeout, Func<bool, Task<TVal>> action)
```

Comment:
You brought back the bool in the Func. Is it really needed?! Because, none of the callers are using it now